src/Microsoft.AspNetCore.Hosting/Internal/StartupLoader.cs

```diff
@@ -117,51 +121,71 @@ public static Type FindStartupType(string startupAssemblyName, string environmen
 
         private static ConfigureBuilder FindConfigureDelegate(Type startupType, string environmentName)
         {
-            var configureMethod = FindMethod(startupType, "Configure{0}", environmentName, typeof(void), required: true);
+            var configureMethod = FindMethod(startupType, Configure, "", environmentName, typeof(void), required: true);
             return new ConfigureBuilder(configureMethod);
         }
 
         private static ConfigureContainerBuilder FindConfigureContainerDelegate(Type startupType, string environmentName)
         {
-            var configureMethod = FindMethod(startupType, "Configure{0}Container", environmentName, typeof(void), required: false);
+            var configureMethod = FindMethod(startupType, Configure, Container, environmentName, typeof(void), required: false);
             return new ConfigureContainerBuilder(configureMethod);
         }
 
         private static ConfigureServicesBuilder FindConfigureServicesDelegate(Type startupType, string environmentName)
         {
-            var servicesMethod = FindMethod(startupType, "Configure{0}Services", environmentName, typeof(IServiceProvider), required: false)
-                ?? FindMethod(startupType, "Configure{0}Services", environmentName, typeof(void), required: false);
+            var servicesMethod = FindMethod(startupType, Configure, Services, environmentName, typeof(IServiceProvider), required: false)
+                ?? FindMethod(startupType, Configure, Services, environmentName, typeof(void), required: false);
             return new ConfigureServicesBuilder(servicesMethod);
         }
 
-        private static MethodInfo FindMethod(Type startupType, string methodName, string environmentName, Type returnType = null, bool required = true)
+        private static MethodInfo FindMethod(Type startupType, string methodNameStart, string methodNameEnd, string environmentName, Type returnType = null, bool required = true)
         {
-            var methodNameWithEnv = string.Format(CultureInfo.InvariantCulture, methodName, environmentName);
-            var methodNameWithNoEnv = string.Format(CultureInfo.InvariantCulture, methodName, "");
-
+            MethodInfo methodInfo = null;
             var methods = startupType.GetMethods(BindingFlags.Public | BindingFlags.Instance | BindingFlags.Static);
-            var selectedMethods = methods.Where(method => method.Name.Equals(methodNameWithEnv)).ToList();
-            if (selectedMethods.Count > 1)
+
+            if (environmentName != null)
             {
-                throw new InvalidOperationException(string.Format("Having multiple overloads of method '{0}' is not supported.", methodNameWithEnv));
+                var selectedMethods = methods.Where(method =>
+                {
+                    if (!method.Name.StartsWith(methodNameStart, StringComparison.Ordinal))
```

Comment:
I thought we didn't want to check against Configure, Services, and Container for case sensitivity. Hence the check on only the env name.